cp-p: init

deadendpl opened the pull request:
## Description of changes

cp-p is a `cp` and `mv` command that shows progress.
It can also be used with lf file manager

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

### Priorities

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

deadendpl committed:
added cp-p

deadendpl committed:
Merge branch 'NixOS:master' into master

hmenke commented on pkgs/tools/misc/cp-p/default.nix:
```diff
@@ -0,0 +1,31 @@
+    homepage = "https://github.com/Naheel-Azawy/cp-p";
+    description = "cp (and mv), with progress";
+    license = with lib.licenses; [ gpl3Only ];
+    maintainers = with lib.maintainers; [ ];
```
No maintainer?

deadendpl replied:
It doesn't seem to have any updates so does it need to have maintainer?

eclairevoyant replied:
you must set yourself as maintainer when adding a new package

hmenke commented on pkgs/tools/misc/cp-p/default.nix:
```diff
@@ -0,0 +1,31 @@
+  meta = {
+    homepage = "https://github.com/Naheel-Azawy/cp-p";
+    description = "cp (and mv), with progress";
+    license = with lib.licenses; [ gpl3Only ];
```
```suggestion
    license = with lib.licenses; [ gpl3Plus ];
```

> If the Program does not specify a version number of the GNU General Public License, you may choose any version ever published by the Free Software Foundation.

eclairevoyant replied:
the readme explicity says GPL3, so I disagree. gpl3Only is fine.

hmenke commented on pkgs/tools/misc/cp-p/default.nix:
```diff
@@ -0,0 +1,31 @@
+, fetchFromGitHub }:
+
+stdenv.mkDerivation {
+  name = "cp-p";
+  pname = "cp-p";
```
```suggestion
  pname = "cp-p";
  version = "unstable-2022-08-07";
```

deadendpl replied:
When I wanted to build that derivation without `name` it didn't want to build

eclairevoyant replied:
name is just pname+version, please make the requested change

hmenke commented on pkgs/tools/misc/cp-p/default.nix:
```diff
@@ -0,0 +1,31 @@
+  # this cancels automatic make
+  dontBuild = true;
+  
+  installPhase = ''
+    mkdir -p $out/bin
+    cp cp-p mv-p $out/bin/
+  '';
```
```suggestion
  installPhase = ''
    runHook preInstall
    mkdir -p $out/bin
    cp cp-p mv-p $out/bin/
    runHook preInstall
  '';
```

That said, it's probably better to use the bundled Makefile and set `makeFlags` such that it does the correct thing.

deadendpl replied:
If such a thing can be achieved then that would be nice

eclairevoyant replied:
mkdir+cp can generally be combined into `install`:
```nix
    install -Dm555 cp-p mv-p -t $out/bin
```

hmenke commented on pkgs/tools/misc/cp-p/default.nix:
```diff
@@ -0,0 +1,31 @@
+    description = "cp (and mv), with progress";
+    license = with lib.licenses; [ gpl3Only ];
+    maintainers = with lib.maintainers; [ ];
+    platforms = lib.platforms.all;
```
```suggestion
    platforms = lib.platforms.all;
```
Isn't this the default?

deadendpl replied:
It probably is. Most of that serivation was inspired by [https://github.com/NixOS/nixpkgs/blob/nixos-23.05/pkgs/tools/misc/dwt1-shell-color-scripts/default.nix#L47](`dwt-shell-color-scripts`)

AndersonTorres replied:
No, it is not.

hmenke reviewed (changes requested):
<strike>This has nothing to do with your packaging, but looking at the upstream source, the shell scripts make a quite poor impression. They claim to be `#!/bin/sh` but then use `function` which is Bash-only. I haven't run shellcheck yet...</strike>

Oops, on further inspection it turns out the snippet in question is an embedded AWK script.

deadendpl commented:
I didn't look at that before. I imported that derivation into my `configuration.nix` and it works as it should.

eclairevoyant commented on pkgs/top-level/all-packages.nix:
```diff
@@ -566,6 +566,8 @@ with pkgs;
 
   copilot-cli = callPackage ../tools/admin/copilot-cli { };
 
+  cp-p = callPackage ../tools/misc/cp-p {};
```
please move to `pkgs/by-name` and remove this

eclairevoyant reviewed (changes requested):
Please ensure your commit messages and PR title follow contributing.md, and always _rebase, never merge_ (merge commit should not be present)

eclairevoyant commented on pkgs/tools/misc/cp-p/default.nix:
```diff
@@ -0,0 +1,31 @@
+    owner = "Naheel-Azawy";
+    repo = "cp-p";
+    rev = "2e97ba534a5892c47a0317a038b19bcda221e5e6";
+    sha256 = "1ki8bp33d9xf4jinr0isvkxq3i86xi1kkv2f5x54as6i0yz9w7iq";
```
please use SRI hash (`hash = "...";`) format

eclairevoyant commented:
Thanks for the contribution! Some changes requested above, and also looks like the CI (ofborg) caught some whitespace issues

hmenke reviewed (changes requested):
The scripts also depend on coreutils to be in `$PATH`, so it's probably best to [resholve](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/misc/resholve/README.md) it.

deadendpl committed:
Update pkgs/tools/misc/cp-p/default.nix

Co-authored-by: Henri Menke <henri@henrimenke.de>

deadendpl committed:
Merge branch 'NixOS:master' into master

deadendpl committed:
Moved cp-p with changes

deadendpl committed:
Update all-packages.nix

deadendpl committed:
Added new maintainer to maintainer-list.nix

deadendpl committed:
Merge branch 'NixOS:master' into master

deadendpl committed:
FIxed maintainer-list.nix

deadendpl commented:
Following some of advices I used `makeFlags` to set variables inside `Makefile` so `installPhase` is now useless.
I also moved my package to `pkgs/by-name`
I also added myself to maintainer list.

If there are any more issues or mistakes please tell me about them.

eclairevoyant commented:
`by-name` uses `package.nix`, not `default.nix`; the old files need to be removed as well.

Additionally the feedback about SRI hash is still to be addressed, as well as the whitespace issues caught by CI

deadendpl committed:
Fixing cp-p

deadendpl committed:
Delete pkgs/tools/misc/cp-p directory

deadendpl committed:
Delete pkgs/by-name/cp/cp-p/default.nix

deadendpl commented:
I changed package file to `default.nix` and removed the old files in `pkgs/tools/misc`.

How can I use SRI Hash and fix whitespace issues?

deadendpl committed:
Merge branch 'NixOS:master' into master

hmenke commented:
> How can I use SRI Hash and fix whitespace issues?

You can convert your hashes into SRI format using the [`nix hash`](https://nixos.org/manual/nix/stable/command-ref/new-cli/nix3-hash) command line utility:
```console
$ nix hash to-sri sha256:1ki8bp33d9xf4jinr0isvkxq3i86xi1kkv2f5x54as6i0yz9w7iq
sha256-OB6evgfRaEVKL07sOUPsBsWB+9w6gmyjJK6nNsZdKM4=
```

Your text editor should have some builtin way to fix whitespace. In, e.g., Emacs it's just `M-x delete-trailing-whitespace`.

```diff
--- package.nix
+++ package.nix
@@ -10,10 +10,10 @@
     owner = "Naheel-Azawy";
     repo = "cp-p";
     rev = "2e97ba534a5892c47a0317a038b19bcda221e5e6";
-    sha256 = "1ki8bp33d9xf4jinr0isvkxq3i86xi1kkv2f5x54as6i0yz9w7iq";
+    hash = "sha256-OB6evgfRaEVKL07sOUPsBsWB+9w6gmyjJK6nNsZdKM4=";
   };
-  
-  makeFlags = [ "PREFIX=$(out)" "BINPREFIX=$(PREFIX)/bin" ]; 
+
+  makeFlags = [ "PREFIX=$(out)" "BINPREFIX=$(PREFIX)/bin" ];
 
   meta = {
     homepage = "https://github.com/Naheel-Azawy/cp-p";
@@ -21,5 +21,5 @@
     license = with lib.licenses; [ gpl3Only ];
     maintainers = with lib.maintainers; [ deadendpl ];
     platforms = lib.platforms.all;
-  }; 
+  };
 }
```

deadendpl committed:
Fixing whitespace in cp-p

deadendpl committed:
Merge branch 'NixOS:master' into master

eclairevoyant commented:
The commits should be squashed into two commits, first one titled `maintainers: add deadendpl` and a second commit with the message `cp-p: init at unstable-2022-08-07`

otherwise looks fine

deadendpl commented:
Is there a way to remove commits or would I need to delete my fork and create it again and then make ready commits to be sent in a new pull request?

eclairevoyant commented:
use `git rebase -i`
also for future reference don't merge from master, if anything you can rebase

deadendpl committed:
Merge branch 'NixOS:master' into master

deadendpl commented:
Can I do `git rebase -i` in a codespace thing?
The repo is too big for my hardware so I was editing everything on github's web page.

deadendpl committed:
Merge branch 'NixOS:master' into master

deadendpl committed:
Merge branch 'NixOS:master' into master

deadendpl commented:
When it comes to rebasing can I delete all of my commits and add 2 complete commits?

deadendpl committed:
Merge branch 'NixOS:master' into master

deadendpl commented:
I tried to change those commits in only 2 but I don't understand `git rebase` at all. How can I achieve this without getting confused?

AndersonTorres commented on pkgs/by-name/cp/cp-p/package.nix:
```diff
@@ -0,0 +1,25 @@
+{ lib
+, stdenv
+, fetchFromGitHub }:
```
```suggestion
, fetchFromGitHub 
}:
```

AndersonTorres reviewed (changes requested):
**Squash**  the commits according to contributing guidelines

donovanglover commented:
Closing due to inactivity. Progress on copy/move can be accomplished with [`rsync`](https://search.nixos.org/packages?channel=unstable&show=rsync), which should be more reliable for critical file operations. It's also [included by default](https://search.nixos.org/options?channel=unstable&show=environment.defaultPackages) in NixOS.
